Sources/Operators/Distinct.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  Distinct.swift
+//  CombineExt
+//
+//  Created by Jasdev Singh on 21/03/2020.
+//  Copyright © 2020 Combine Community. All rights reserved.
+//
+
+import Combine
+
+public extension Publisher where Output: Hashable {
+    /// De-duplicates _all_
+    /// published value events, as opposed to pairwise with `Publisher.removeDuplicates`.
+    ///
+    /// - note: It’s important to note that this operator stores all emitted values
+    ///         in an in-memory `Set`. So, use this operator with caution, when handling publishers
+    ///         that emit a large number of unique value events.
+    ///
+    /// - returns: A publisher that consumes duplicate values across all previous emissions from upstream.
+    func distinct() -> Publishers.Filter<Self> {
+        var seen = Set<Output>()
```

Comment:
There's something I hadn't considered. A set determines uniqueness by Hasability. That's why the definition of it is

`public struct Set<Element> where Element: Hashable {`

Wouldn't this basically mean you can't use this operator for anything that isn't Hashable-cconforming?

Reply:
That’s where the more relaxed `Equatable` and comparator-based overloads come in. :nodg: